### ISSUE TITLE
NGOptRW stuck on high num_workers

### ISSUE THREAD

**Issue by mstniy** [facebookresearch/nevergrad#1506]
## Steps to reproduce

The setup is rather brittle. Using >~9 workers, NGOptRW and 4 keyword arguments breaks nevergrad completely.

Python version: 3.10.6
Nevergrad version: 0.6.0

## Observed Results

The optimizer freezes.

## Expected Results

For the optimizer to find a solution.

## Relevant Code

```
import nevergrad as ng


def _loss(*args, **kwargs):
    print("_loss:", args, kwargs)
    return 0


def main():
    parametrization = ng.p.Instrumentation(
        a=ng.p.Scalar(lower=1, upper=10),
        b=ng.p.Scalar(lower=1, upper=10),
        c=ng.p.Scalar(lower=1, upper=10),
    )

    optimizer = ng.optimizers.NGOptRW(
        parametrization=parametrization, budget=1000, num_workers=9
    )
    optimizer.minimize(
        _loss,
        verbosity=2,
    )


if __name__ == "__main__":
    main()

```

**Comment by Obliman** [comment]
I tried this with a few different optimizers for comparison (same python and nevergrad versions):

- NGOptRW gets stuck on startup with the message "Launching 9 jobs with new suggestions" followed by nothing happening, or the print statement being evaluated in the _loss function for 8 of the 9 workers. The print statement is also only evaluated 8 times when the number of workers is increased further to 16, for example.
- NGOpt, NoisyBandit, CMA, and OnePlusOne perform as expected without warnings/errors
- CMandAS3 runs, but with many Injection Warnings:
```
InjectionWarning: orphanated injected solution {'iteration': 1, 'index': 0, 'counter': 0}
                        This could be a bug in the calling order/logics or due to
                        a too small popsize used in `ask()` or when only using
                        `ask(1)` repeatedly. Please check carefully.
                        In case this is desired, the warning can be surpressed with
                        ``warnings.simplefilter("ignore", cma.evolution_strategy.InjectionWarning)``
```
(Usually 'index'=0 and  'counter'=0 to 2)

Additionally, I found that with three or fewer workers NGOptRW could run the analysis as expected, but with four workers it would suddenly freeze part way though.

**Comment by Obliman** [comment]
FYI, I found more bugs related to CMandAS3, which has become its own issue (#1508).

**Comment by teytaud** [comment]
I've reproduced this issue with Nevergrad 0.6.0.
Investigating now and testing with more recent Nevergrad.

**Comment by teytaud** [comment]
Thank you very much. 
The issue is understood and fixed in https://github.com/facebookresearch/nevergrad/pull/1555
Will be merged asap and in release 0.13.0.

**Comment by teytaud** [comment]
Should be solved in the current main and in 0.13.0 which is on its way to release on Pypi.
